libdrgn/object.c

```diff
@@ -110,7 +110,8 @@ drgn_object_type_impl(struct drgn_type *type, struct drgn_type *underlying_type,
 						 "bit field must be integer");
 		}
 		if (ret->encoding == DRGN_OBJECT_ENCODING_FLOAT &&
-		    ret->bit_size != 32 && ret->bit_size != 64) {
+		    ret->bit_size != 32 && ret->bit_size != 64 &&
+		    ret->bit_size != 80 && ret->bit_size != 128) {
```

Comment:
It looks like [GCC](https://gcc.gnu.org/onlinedocs/gcc/Floating-Types.html) and [Clang](https://clang.llvm.org/docs/LanguageExtensions.html#half-precision-floating-point) both also have a 16-bit floating-point type. Maybe rather than allowing a few specific sizes here, we can be permissive and allow anything within a "reasonable" range here like we do for integers (perhaps 1-256?). Note that this means that the checks for value creation would have to check `bit_size != 32 && bit_size != 64` instead of `bit_size > 64`.

Reply:
Sounds good, I modified it to allow 1-256, as you mentioned. Conditions, error messages, and test-cases have been updated accordingly.